src/de/grobox/liberario/activities/MainActivity.java

```diff
@@ -205,6 +210,21 @@ public void onDrawerSlide(View drawerView, float slideOffset) {
 	}
 
 	@Override
+	protected void onResume() {
+		super.onResume();
+		// check if there's a preference for selected products for the current network
+        TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+        SharedPreferences sp = getContext().getSharedPreferences("pref_selected_products", 0);
+        String network_pref = current_network + "_selected_products";
+        if(sp.contains(network_pref)){
+            // get the selected products from the pref
+            Set<String> product_set = sp.getStringSet(network_pref, new HashSet<String>());
+            //TODO: initiate the loading of the products from the product_set?
+            //where's actually the point, where the products are loaded?
```

Comment:
You don't need to touch `MainActivity` at all. Look into the `DirectionsFragment`. That's the only place where the products are used at the moment.